warehouse/admin/views/checks.py

```diff
@@ -67,18 +109,16 @@ def get_check(request):
     require_csrf=True,
 )
 def change_check_state(request):
+    check = get_check_by_name(request.db, request.matchdict["check_name"])
+
```

Comment:
For viewing... fetching by name seems reasonable to me, but for submitting a state change fetching by ID seems preferable. Could consider a final sanity check that the check being modified is the latest?

Reply:
I did it this way because I view it as safer. A user of the web admin should never be able to craft a request that modifies the state of an arbitrary check via `id`. This way, the view logic determines which check is the correct entry to modify.